catalog/files/impl/src/main/java/org/projectnessie/catalog/files/adls/AdlsClientSupplier.java

```diff
@@ -92,16 +93,31 @@ DataLakeFileSystemClient fileSystemClient(AdlsLocation location) {
                             location.storageAccount(),
                             location.container().map(c -> ", container " + c).orElse("")))));
 
-    if (fileSystemOptions.sasToken().isPresent()) {
-      clientBuilder.sasToken(fileSystemOptions.sasToken().get().key());
-    } else if (fileSystemOptions.account().isPresent()) {
-      String accountKey = fileSystemOptions.account().get().secret();
-      clientBuilder.credential(new StorageSharedKeyCredential(accountName, accountKey));
-    } else {
-      throw new IllegalStateException(
-          "Neither an SAS token nor account name and key are available for ADLS file system '"
-              + location.path()
-              + "'");
+    AdlsFileSystemOptions.AzureAuthType authType =
+        fileSystemOptions
+            .authType()
+            .orElse(AdlsFileSystemOptions.AzureAuthType.APPLICATION_DEFAULT);
```

Comment:
Guess this should default to `NONE` and let the user explicitly define the type.

Reply:
@snazy, just tested the behavior when I don't call `clientBuilder.credential`, I will get the following error: 
```
"com.azure.storage.file.datalake.models.DataLakeStorageException: Status code 401, \"﻿<?xml version=\"1.0\" encoding=\"utf-8\"?><Error><Code>NoAuthenticationInformation</Code><Message>Server failed to authenticate the request. Please refer to the information in the www-authenticate header.\nRequestId:22377f67-e01e-0060-54g2-d9e145123456\nTime:2024-07-19T15:35:55.3953452Z</Message></Error>\""
```
which to me is correct behavior since no authentication information is provided. I will proceed to add in the `NONE` enum and default to it

Reply:
I feared that it behaves like that, but it looks like you can use `com.microsoft.azure.storage.StorageCredentialsAnonymous#ANONYMOUS` in the `NONE` case - it's probably not that useful in practice, but who knows.

Reply:
I think I can set Anonymous access in the `NONE` case by `clientBuilder.setAnonymousAccess();`. I will push the changes, let me know if it is ok.